components/common/Tab.js

```diff
@@ -0,0 +1,137 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+import { Link } from 'routes';
+import TetherComponent from 'react-tether';
+
+// components
+import SubMenu from 'components/common/SubMenu';
+
+export default class Tab extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      category: ''
+    };
+
+    // saves node references of tabs to colocate submenus
+    this.tabNodes = [];
+  }
+
+  onClickTab(e, category) {
+    if (e) e.preventDefault();
+    this.setState({ category });
+  }
+
+  onCloseSubMenu() {
+    this.setState({
+      category: ''
+    });
+  }
+
+  renderTab(item, index) {
+    const { allowAll, queryParams } = this.props;
+    const { route, category } = queryParams;
+    const { children, label, query } = item;
+
+    if (children) {
+      let subMenuOptions = children;
+
+      if (allowAll) {
+        subMenuOptions = [
+          ...[{ label: 'See all', query: { category: query.category } }],
+          ...children
+        ];
+      }
+
+      return (
+        <TetherComponent
+          attachment="top center"
+          targetAttachment="top center"
+          targetOffset="-15px 0"
+          key={label}
+          constraints={[{
+            to: 'target',
+            attachment: 'together'
+          }]}
+        >
+          <li
+            ref={(node) => { this.tabNodes[index] = node; }}
+            className={classnames('tab-item', { '-current': query.category === category })}
+            role="menuitem"
+            aria-haspopup="true"
+            tabIndex="-1"
+          >
+            <a
+              href="explore"
```

Comment:
It's a way to avoid eslint telling me to not set '#' or whatever in `href` attribute 😂